File: AVG.py
# cook your dish here
for _ in range(int(input())):
    n,k,v=map(int,input().split())
    A=list(map(int,input().split()))[:n]
    ans = sum(A)
    if v*(n+k)-ans<=0:
        print("-1")
    elif (v*(n+k)-ans)%k==0:
        print((v*(n+k)-ans)//k)
    else:
        print("-1")

File: AVG.cpp
#include <iostream>
using namespace std;

int main() {
	// your code goes here
	int t;
	cin>>t;
	while(t--){
	    int n,k,v;
	    cin>>n>>k>>v;
	    long long sum=0;
	    int ar[n];
	    for(int i=0;i<n;i++){
	        cin>>ar[i];
	        sum=sum+ar[i];
	    }
	    int nr=((v*(n+k))-sum);
	    int ans=nr/k;
	    
	    if(ans>0 and nr%k==0){
	        cout<<ans<<endl;
	    }
	    else{
	        cout<<"-1"<<endl;
	    }
	}
	
	return 0;
}
